lib/streamlit/runtime/caching/cache_utils.py

```diff
@@ -80,6 +80,11 @@ def ttl_to_seconds(
         return math.inf
     if isinstance(ttl, timedelta):
         return ttl.total_seconds()
+    if isinstance(ttl, str):
+        import pandas as pd
+
+        return pd.Timedelta(ttl).total_seconds()
```

Comment:
nit: we could theoretically throw the `ValueError` as a `StreamlitAPIException`. We are not super consistent with this across the codebase. But since the `ttl` is only used as part of an `st` API, it would make it a bit more explicit that we expect that an error can occur here.

```python
try:  
    return pd.Timedelta(ttl).total_seconds()
except ValueError as ex:
    raise StreamlitAPIException('The ttl parameter %s is invalid' % ttl) from ex
```

Also, if the string is empty, we should probably return `None` here since `pd.Timedelta("").total_seconds()` will return `np.nan`.